### DISCUSSION TITLE
Pages per visit

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#131] [Anything Else]
"Pages seen per user. I can see at the top of the chart “unique visitors” and “total pageviews” it would be nice to add the line (or allow me to add) with “average pages visited per user” (totalPageViews divided by uniqueVisitors).
Especially on the multi-day charts this might be helpful. Thanks!!!
Good job with the tool. Keep it up!"

**Update:** You can now find two brand new metrics in your dashboard: "Total visits" (sessions) and "Views per visit" (pageviews per visit/session).

**Comment by bjoernbg** [comment]
This would be helpful for us as well. How many pages a visitor visits is a key metric for us!

**Comment by kristjanmar** [comment]
I agree with this, it is one of the core engagement metrics that should be included along with bounce rate and visit duration.

**Comment by flaviolivolsi** [comment]
This would be really nice, I came here to submit this feature and I found it here. Do you know whether it will be implemented?

**Reply by metmarkosaric** [reply]
hi @flaviolivolsi! we're working on adding the number of visits (sessions) to the dashboard right now. after that is completed, we're going to work on adding the number of pages per visit. cannot make any promise on when it will be ready to go live but we're on it. stay tuned!

**Reply by flaviolivolsi** [reply]
Amazing @metmarkosaric - thank you!

**Comment by metmarkosaric** [comment]
**Update**: You can now find two brand new metrics in your dashboard: "Total visits" (sessions) and "Views per visit" (pageviews per visit/session).

**Comment by wuerzi** [comment]
Why is the feature „views per visit“ in our selfhosting-version still missing? Is there any switch to activate this feature?

**Reply by ruslandoga** [reply]
👋 @wuerzi 

Seems like this feature is still hidden by a feature flag:

https://github.com/plausible/analytics/blob/6c907cbf4221788b31e78de77a26d32440feadf2/lib/plausible_web/controllers/stats_controller.ex#L330

You can try enabling it manually with

```console
$ cd hosting
$ docker compose exec plausible bin/plausible rpc 'FunWithFlags.enable(:views_per_visit_metric)'
```

**Reply by ruslandoga** [reply]
https://github.com/plausible/analytics/discussions/2896#discussioncomment-5801977 might be relevant as well if you are on `v2.0.0-rc.0`

**Reply by wuerzi** [reply]
Aah i see. Thank you. So i have to wait, because i am using the self-hosting-Version which isnt probably available at `v2.0.0-rc.0`

**Reply by ruslandoga** [reply]
`v2.0.0-rc.0` is a self-hosted [pre-prelease.](https://github.com/plausible/analytics/discussions/2896) You can try it out, but note the data migration steps :)

And you can also try `v2.0.0-rc.1`: https://github.com/plausible/analytics/discussions/2981

**Reply by wuerzi** [reply]
We will try - Thanl you.
But as we use it in a production-environment, we still have to wait. 👍
